services/httpd/cursor.md

```diff
@@ -0,0 +1,162 @@
+
+## Protocol Format
+
+The protocol is made up of a stream of objects. The first object returned
+back from the server is the `Response`. The `Response` contains any
+metadata about the response itself.
+
+    type Response struct {
+        // Results number of results that will be sent to this cursor.
+        Results int
+        // Error contains any error while processing the query,
+        // but before any results are processed.
+        Error   string
+    }
+
+If there is an error returned with the `Response`, the stream is finished.
+The results field is the number of expected results from this query. There
+is one expected result from each statement.
+
+Each result starts with a `Result`.
+
+    // Result is the metadata of the current result.
+    type Result struct {
+        ID int
+        Messages []Message
+        Error string
+    }
+
+    // Message is a nested data structure for an informational message
+    // sent back by the server for a query.
+    type Message struct {
+        Level string
+        Text string
+    }
+
+It contains the result id which will match with the result number
+(zero-indexed). Embedded are any informational messages that can be
+displayed to the user such as deprecation messages. If an error is
+encountered, read the next result.
+
+If there is no error, then the next object will be `NextSeries`.
+
+    // NextSeries contains information about if there is another series
+    // to be read from the result.
+    // When encoded as MessagePack, this is a boolean.
+    type NextSeries struct {
+        // Present is true if there is another series to read.
+        Present bool
+    }
+
+`NextSeries` communicates whether there is a next series or if we have
+finished reading the result. If true, then a series should be read.
+
+    type Series struct {
+        // Name contains the measurement name of this series.
+        Name string
+
+        // Tags contains a mapping of tag keys to tag values for this
+        // series.
+        Tags map[string]string
+
+        // Columns contains the column names for this series.
+        Columns []string
+
+        // Error is set if an error was encountered while processing
+        // this series.
+        Error string
+    }
+
+This contains the series metadata. If the error is set, then skip the
+following and check for `NextSeries` again. If there is no error,
+then read the `RowBatchHeader`.
+
+    // RowBatchHeader contains information about the current batch.
+    type RowBatchHeader struct {
+        // Length contains the number of rows that are in this batch.
+        Length int
+
+        // Continue is set to true if there is another batch that
+        // should be read.
+        Continue bool
+
+        // Error is present if an error was encountered while
+        // emitting the rows for the query.
+        Error string
+    }
+
+The `RowChunkHeader` contains metadata for the rows that are about to be
```

Comment:
Yea, I changed the name several times. I hate all of the names. Feel free to give feedback on them. I'm about ready to name them using `/dev/urandom` at this point.